rewrite-maven/src/main/java/org/openrewrite/maven/AddOrUpdateDependencyManagement.java

```diff
@@ -0,0 +1,291 @@
+                dependencies.sort(Comparator.comparing(d -> d.getModel().getModuleVersion()));
+
+                int addAfterIndex = -1;
+                for (int i = 0; i < sortedDependencies.size(); i++) {
+                    Maven.Dependency d = sortedDependencies.get(i);
+                    if (toAdd == d) {
+                        addAfterIndex = i - 1;
+                        break;
+                    }
+                }
+                dependencies.add(addAfterIndex + 1, toAdd);
+            }
+            p = p.withDependencyManagement(p.getDependencyManagement().withDependencies(dependencies));
+            return p;
+        }
+    }
+
+    private class UpdateDependency extends MavenRefactorVisitor {
+        @Override
+        public Maven visitPom(Maven.Pom pom) {
+            Maven.Pom p = refactor(pom, super::visitPom);
+            List<Maven.Dependency> dependencies = new ArrayList<>(pom.getDependencyManagement().getDependencies());
+            Maven.Dependency dependency = findDependencyWithGroupIdAndArtifactId(dependencies, groupId, artifactId);
+
+            final Xml.Tag tag = createDependencyTag(formatter.wholeSourceIndent());
+            final Maven.Dependency updatedDependency = new Maven.DependencyManagement.Dependency(false,
+                    new MavenModel.Dependency(
+                            new MavenModel.ModuleVersionId(groupId, artifactId, null, version, "jar"),
+                            version,
+                            scope
+                    ),
+                    tag
+            );
+
+            dependencies.set(dependencies.indexOf(dependency), updatedDependency);
+            p = p.withDependencyManagement(p.getDependencyManagement().withDependencies(dependencies));
+            return p;
+        }
+
+        private Maven.Dependency findDependencyWithGroupIdAndArtifactId(List<Maven.Dependency> dependencies, String groupId, String artifactId) {
+            return dependencies.stream()
+                    .filter(d -> groupId.equals(d.getGroupId()) && artifactId.equals(d.getArtifactId()))
+                    .findFirst().get();
+        }
+
+        private int getIndexOfDependencyToBeRemoved(List<Maven.Dependency> dependencies) {
+            final Optional<Maven.Dependency> optionalDependency = dependencies.stream()
+                    .filter(d -> groupId.equals(d.getGroupId()) && artifactId.equals(d.getArtifactId()))
+                    .findFirst();
+            return dependencies.indexOf(optionalDependency.get());
+        }
+
+        private void removeDependency(List<Maven.Dependency> dependencies) {
+            final Optional<Maven.Dependency> optionalDependency = dependencies.stream()
+                    .filter(d -> groupId.equals(d.getGroupId()) && artifactId.equals(d.getArtifactId()))
+                    .findFirst();
+            final Maven.Dependency dependency = optionalDependency.get();
+            dependencies.remove(dependency);
+        }
```

Comment:
Looks like these methods aren't being used. Are they safe to delete or is there some functionality you're thinking of adding that depends on them?

Reply:
Thanks, indeed. I removed them